tracing-subscriber/src/subscribe/mod.rs

```diff
@@ -1475,6 +1475,26 @@ pub struct Identity {
 
 // === impl Subscribe ===
 
+#[derive(Clone, Copy)]
+pub(crate) struct NoneLayerMarker;
+pub(crate) static NONE_LAYER_MARKER: NoneLayerMarker = NoneLayerMarker;
```

Comment:
nit, not a big deal: it might be nice to give the marker type a single private field, so that it can't be created by code outside this file? that would help ensure that the static `NONE_LAYER_MARKER` is the _only_ instance of this type that's ever instantiated, which ensures that the pointers to it that are returned are actually always valid.

this doesn't _matter_ a whole lot, since no one will ever actually dereference a pointer to this thing, but minimizing the potential of accidentally creating an invalid reference still makes me feel more comfortable...
```suggestion
pub(crate) struct NoneLayerMarker(());
pub(crate) static NONE_LAYER_MARKER: NoneLayerMarker = NoneLayerMarker(());
```